### PR TITLE
Allowing images to be more easily imported from node_modules

### PR DESCRIPTION
**Pull request by stephencookdev**
This will allow the following sort of thing in a pinionfile:

```
images: {
  src: [
    "base/img/path",
    "some_npm_module/imgs"
  ],
  dest: ""
}
```

### CODE REVIEW

File: tasks/images.js [hunk 1]
```diff
@@ -12,8 +12,18 @@ var imagemin = require('../lib/gulpImagemin');
 module.exports = function(config) {
   if(!config.tasks.images) return;
 
+  var relativePath = config.tasks.images.src;
+  if(Array.isArray(config.tasks.images.src)) {
```
Comment (rfreitas) [inline tasks/images.js]: Minor: why not `if(Array.isArray(relativePath))` ?

File: tasks/images.js [hunk 2]
```diff
@@ -12,8 +12,18 @@ var imagemin = require('../lib/gulpImagemin');
 module.exports = function(config) {
   if(!config.tasks.images) return;
 
+  var relativePath = config.tasks.images.src;
```
Comment (rfreitas) [inline tasks/images.js]: Minor: `relativePath` could have a mention that is related to images

File: tasks/images.js [hunk 3]
```diff
@@ -12,8 +12,18 @@ var imagemin = require('../lib/gulpImagemin');
...
+  }
+
   var paths = {
-    src: path.join(config.root.src, config.tasks.images.src, '/**'),
+    src: [
+      path.join(config.root.src, relativePath, '/**'),
+
+      // Search for images in the package's node_modules too
+      path.join(process.cwd(), 'node_modules', relativePath, '/**')
+    ],
```
Comment (rfreitas) [inline tasks/images.js]: so the paths in:
```js
images: {
  src: [
    "base/img/path",
    "some_npm_module/imgs"
  ],
  dest: ""
}
```

just to make sure I got it right, the paths are relative to the `root` and the `node_modules` at the same time, correct?

Comment (stephencookdev) [inline tasks/images.js]: Yeah, so you will get a collision if `some_npm_module/some/path/img.png` and `base/some/path/img.png` both exist

I'm just assuming that this shouldn't really be an issue

Comment (rfreitas) [inline tasks/images.js]: does it resolve the collision in favour of the project root? If so collisions can be worked around easily.

Comment (stephencookdev) [inline tasks/images.js]: Good point, it did not have that ensured - no.

That should now be ensured. For the src array `[x, y, z]`, the search array `[root/x, npm/x, root/y, npm/y, root/z, npm/z]` will be generated - with priority being the sooner in the array, the higher the priority.

**Comment by rfreitas** [comment]
looks good ;)

**Comment by rfreitas** [comment]
btw, should these changes be documented in the `READMEN.MD` file?

**Comment by stephencookdev** [comment]
Yes, these should be documented in the `README.md`, good shout

I'll add the documentation before merging

**Pull request merged**